discovery/scaleway/instance.go

```diff
@@ -175,14 +177,42 @@ func (d *instanceDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group,
 		}
 
 		addr := ""
+		if len(server.PublicIPs) > 0 {
+			var ipv4Addresses []string
+			var ipv6Addresses []string
+
+			for _, ip := range server.PublicIPs {
+				if ip.Family == instance.ServerIPIPFamilyInet {
+					ipv4Addresses = append(ipv4Addresses, ip.Address.String())
+				} else if ip.Family == instance.ServerIPIPFamilyInet6 {
+					ipv6Addresses = append(ipv6Addresses, ip.Address.String())
+				}
+			}
+
+			if len(ipv6Addresses) > 0 {
+				labels[instancePublicIPv6AddressesLabel] = model.LabelValue(
+					separator +
+						strings.Join(ipv6Addresses, separator) +
+						separator)
+			}
+			if len(ipv4Addresses) > 0 {
+				labels[instancePublicIPv4AddressesLabel] = model.LabelValue(
+					separator +
+						strings.Join(ipv4Addresses, separator) +
+						separator)
+			}
+		}
+
 		if server.IPv6 != nil { //nolint:staticcheck
 			labels[instancePublicIPv6Label] = model.LabelValue(server.IPv6.Address.String()) //nolint:staticcheck
 			addr = server.IPv6.Address.String()                                              //nolint:staticcheck
 		}
 
 		if server.PublicIP != nil { //nolint:staticcheck
-			labels[instancePublicIPv4Label] = model.LabelValue(server.PublicIP.Address.String()) //nolint:staticcheck
-			addr = server.PublicIP.Address.String()                                              //nolint:staticcheck
+			if server.PublicIP.Family != instance.ServerIPIPFamilyInet6 { //nolint:staticcheck
```

Comment:
I understand, but I'm afraid a long-lived bug can become a feature as they say, maybe we're relying on it for relabeling or sth even when it yields an IPv6.

We do not guarantee any stability for those things, but if we can avoid "breaking" changes when possible it'd be great.

Let's just add a comment around that code and let's fix if someone (other than us) screams out of it or once `PublicIP` and `IPv6` fields disappear, whichever of the two comes first.